client/state/analytics/actions.js

```diff
@@ -50,12 +57,25 @@ export const recordEvent = ( service, args ) => ( {
 	}
 } );
 
+export const recordEventOnce = ( eventId, service, args ) => ( dispatch, getState ) => {
+	const state = getState();
```

Comment:
By the way, in envisioning global selectors (#5954), `hasAlreadyBeenRecorded` is not a particularly unique or identifiable name for a selector. Maybe `hasEventAlreadyBeenRecorded` instead.